src/Rings/mpoly-ideals.jl

```diff
@@ -1572,6 +1572,90 @@ function equidimensional_hull_radical(
   return ideal(R, unique!(iso.(gens(res))))
 end
 
+################################################################################
+#
+#  triangular decomposition
+#
+################################################################################
+
+@doc raw"""
+    triangular_decomposition(I::MPolyIdeal; algorithm::Symbol=:lazard)
+
+Return a triangular decomposition of a zero-dimensional `I`.
+
+Possible algorithms are:
+- `:lazard` (default) based on [Laz92](@cite).
+- `:lazard_factorized` further factorizes each polynomial in the output of `:lazard`.
+- `:moeller` based on [Moe93](@cite)
+- `:moeller_hillebrand` based on [Hil99](@cite).
+
+# Examples
+```jldoctest
+julia> R, (x1,x2,x3,x4,x5) = polynomial_ring(QQ, [:x1,:x2,:x3,:x4,:x5]);
+
+julia> I = ideal([x1+x2+x3+x4+x5,
+                  x1*x2+x2*x3+x3*x4+x1*x5+x4*x5,
+                  x1*x2*x3+x2*x3*x4+x1*x2*x5+x1*x4*x5+x3*x4*x5,
+                  x1*x2*x3*x4+x1*x2*x3*x5+x1*x2*x4*x5+x1*x3*x4*x5+x2*x3*x4*x5,
+                  x1*x2*x3*x4*x5-1]);
+
+julia> triangular_decomposition(I)
+4-element Vector{MPolyIdeal{QQMPolyRingElem}}:
+ Ideal (x5^5 - 1, x4 - x5, x3^2 + 3*x3*x5 + x5^2, x2 + x3 + 3*x5, x1 - x5)
+ Ideal (x5^5 - 1, x4 - x5, x3 - x5, x2^2 + 3*x2*x5 + x5^2, x1 + x2 + 3*x5)
+ Ideal with 5 generators
+ Ideal with 5 generators
+
+julia> triangular_decomposition(I, :LazardFactorized)
```

Comment:
```suggestion
julia> triangular_decomposition(I; algorithm = :lazard_factorized)
```
and also in the rest of the docstring